### PR TITLE
context changes auth n crunch service

### PR DESCRIPTION
**Pull request by Nauna**
start update for using Auth.sudo when we are updating a context to a different user...

### CODE REVIEW

**Comment by jlhughes** [comment]
@Nauna can you think of reason to not set the group by default? 
So  sudo(x, user) calls -> sudo(x, user, user.findGroup(x))

**Comment by Nauna** [comment]
> @Nauna can you think of reason to not set the group by default? So sudo(x, user) calls -> sudo(x, user, user.findGroup(x))

no not really - I like this simplification ... I'll quickly go through the other calls in the code see if there is anything Im missing then make this update...30 min - moving pr to dnm

File: src/foam/nanos/crunch/ServerCrunchService.java [hunk 1]
```diff
@@ -203,7 +202,10 @@ public String[] getDependentIds(X x, String capabilityId) {
   // gets prereq list of a cap from the prereqsCache_
   // if cache returned is null, try to find prereqs directly from prerequisitecapabilityjunctiondao
   public List<String> getPrereqs(X x, String capId, UserCapabilityJunction ucj) {
-    if ( ucj != null ) x = x.put("subject", ucj.getSubject(x));
+    if ( ucj != null ) {
```
Comment (Nauna) [inline src/foam/nanos/crunch/ServerCrunchService.java:205]: try with removing

File: src/foam/nanos/crunch/Capability.js [hunk 2]
```diff
@@ -505,9 +488,9 @@ foam.CLASS({
             var cap = (Capability) capabilityDAO.find(capId);
             if ( cap == null || cap.getLifecycleState() != foam.nanos.auth.LifecycleState.ACTIVE ) continue;
 
-            UserCapabilityJunction prereqUcj = crunchService.getJunctionForSubject(x, capId, subject);
+            UserCapabilityJunction prereqUcj = crunchService.getJunction(x, capId);
```
Comment (noodlemoodle) [inline src/foam/nanos/crunch/Capability.js]: does this need to be `getJunctionForSubject(x, capId, ucj.getSubject(x));` ?

**Comment by noodlemoodle** [comment]
one comment - rest lgtm

File: src/foam/nanos/crunch/Capability.js (hunk 2 above)
Comment (noodlemoodle) [reply src/foam/nanos/crunch/Capability.js]: updated in https://github.com/kgrgreer/foam3/pull/1623/commits/65c8a43d6d0d1d3e6b9eb621b20942eb1b240dcd
changing this fixes intuit test

**Review by noodlemoodle — APPROVED** [review]

**Pull request merged**